test/dhtest.c

```diff
@@ -166,6 +166,9 @@ int main(int argc, char *argv[])
 
     alen = DH_size(a);
     abuf = OPENSSL_malloc(alen);
+    if (!abuf)
```

Comment:
the way to check empty point has been modified as the way you advised ( change (!abuf)  to----> (abuf == NULL) ), thanks for your advise:)